vnext/Desktop/WinHTTPWebSocketResource.h

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+#pragma once
+
+#include <IWebSocketResource.h>
+
+// Windows API
+#include <winhttp.h>
+
+// Standard Library
+#include <functional>
+
+namespace Microsoft::React
+{
+class WinHTTPWebSocketResource : public IWebSocketResource
+{
+  static VOID CALLBACK Callback(HINTERNET handle, DWORD_PTR context, DWORD status, LPVOID info, DWORD infoLength);
+
+  URL_COMPONENTS m_url;
+  DWORD m_openSessionFlags;
+  HINTERNET m_sessionHandle;
+  HINTERNET m_connectionHandle;
+  HINTERNET m_requestHandle;
+  HINTERNET m_webSocketHandle;
+
+  DWORD m_errorStatus;
```

Comment:
>m_errorStatus [](start = 8, length = 13)

rename to m_lastError since this is always set based on GLE (not other types of errors)